internal/controller/worker_controller_test.go

```diff
@@ -29,152 +39,391 @@ var (
 			},
 		},
 	}
+
+	testTemporalNamespace = "ns"
 )
 
+// Hash is always taken of Spec.Template for generating the buildID.
 func newTestWorkerSpec(replicas int32) *temporaliov1alpha1.TemporalWorkerDeploymentSpec {
 	return &temporaliov1alpha1.TemporalWorkerDeploymentSpec{
 		Replicas: &replicas,
 		Template: testPodTemplate,
 		WorkerOptions: temporaliov1alpha1.WorkerOptions{
-			TemporalNamespace: "baz",
+			TemporalNamespace: testTemporalNamespace,
 		},
 	}
 }
 
-func newTestDeployment(podSpec v1.PodTemplateSpec, desiredReplicas int32) *appsv1.Deployment {
+// newTestDeployment creates a new Deployment object for testing purposes.
+// If buildID is empty, it will be computed from the desiredState.Template.
+func newTestDeployment(podSpec v1.PodTemplateSpec, desiredState *temporaliov1alpha1.TemporalWorkerDeploymentSpec, deploymentName string, buildID string) *appsv1.Deployment {
+	// If buildID is not provided, compute it from the template
```

Comment:
this looks like a good solution to me! and avoids having to maintain repeated code 👍